students/roviram/lesson05/mailroom_part_3.py

```diff
@@ -0,0 +1,151 @@
+
+
+def thank_you():
+    """Thanking the donor for their donor and adding a new donor if they are new"""
+    list_answer = ""
+    while True:
+        while True:
+            list_answer = input("\nWant to see the list of donors to send a 'Thank You' too? (type 'Yes' or 'No ') "
+                                ).lower().strip()
+
+        # Cleaning up the Users Answer
+            if list_answer == 'yes':
+                for donor in donors:
+                    print(donor)
+                break
+            elif list_answer == 'no':
+                break
+            else:
+                continue
+
+        donor_name = input("\nPlease enter the donor name: ")
+
+        # Prompt Donor Name
+        print("You typed:", donor_name)
+        match = False
+
+        if donor_name in donors:
+            print("Cool, I found that donor")
+            match = True
+
+        if not match:
+            print("I didn't find that donor, I am adding them to our database now.")
+            donors[donor_name] = []
+        while True:
+            try:
+                donation_amount = float(input("\nHow much did they donate?: "))
+                break
+            except ValueError:
+                print("Please enter a valid number.")
+
+        donors[donor_name].append(donation_amount)
+        print("\nTime to send a thank you email!")
+        print(f"""Dear {donor_name},
+        
+Thank you for your generous donation of ${donation_amount}.
+Sincerely,
+
+The Mailroom Team""")
+
+        menu_answer = input("\nAre you done? ('Yes' or 'No' ) ").upper().strip()
+        if menu_answer == 'YES':
+            break
+
+
+def send_letters(donors_dict):
+    """
+    :param donor_name: name of the donor in our database
+    :param donation_amount: the amount the donor donated
+    """
+    for donor in donors_dict:
+        with open(donor + ".txt", "w") as file_output:
+            file_output.write(f"""Dear {donor},
+Thank you for your most recent generous donation of ${donors_dict[donor][-1]}.
+
+Sincerely,
+
+The Mailroom Team""")
+
+#".format(donor, donors_dict[donor][-1]))
+    print("\nWe have sent letters to all the donors! ")
+
+
+def main_menu():
+    """This is displaying the main menu"""
+    while True:
+        try:
+            menu_answer = int(input(
+                "\nPlease choose from the following three options (please type 1, 2, 3, or 4):\n"
+                "1) Send a 'Thank You' note to a single donor \n"
+                "2) Create a Report \n"
+                "3) Send letters to all donors \n"
+                "4) Quit "))
+
+    # Thank You Note
+            if menu_answer == 1:
+                thank_you()
+                # Create a Report
+            elif menu_answer == 2:
+                create_report(donors)
+            elif menu_answer == 3:
+                send_letters(donors)
+            elif menu_answer == 4:
+                break
+        except ValueError:
```

Comment:
you want to catch the exception as close as you can to where it might happen -- so in this case, right after the call to int(). Otherwise, it could catch that same exception from practically anywhere!